tests/rust/test_rust_backend.py

```
@@ -579,75 +648,88 @@ def test_backend_state_consistency(self, rust_backend, sample_rust_code, temp_di
     backend = RustBackend()
     source_file = temp_dir / f"{expected_behavior}.rs"
     source_file.write_text(rust_code)
-    if hasattr(backend, 'compile'):
+    if hasattr(backend, "compile"):
         try:
             result = backend.compile(str(source_file))
             if expected_behavior == "simple_success":
                 assert result is not None
             elif expected_behavior == "compile_error":
-                if hasattr(result, 'success'):
+                if hasattr(result, "success"):
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)